packages/network/test/unit/Speedometer.test.ts

```diff
@@ -0,0 +1,129 @@
+import { Speedometer } from '../../src/helpers/Speedometer'
+
+const WINDOW_SIZE_IN_SECONDS = 5
+
+const createMockCurrentTimeProvider = () => {
+    let now: number
+    const fn = () => now
+    fn.set = (dateStr: string) => now = new Date(dateStr).getTime()
+    return fn
+}
+
+const STARTUP_TIME = '2000-01-02T00:00:00.123Z' 
+
+describe('Speedometer', () => {
+
+    let speedometer: Speedometer
+    const currentTimeProvider = createMockCurrentTimeProvider()
+
+    beforeEach(() => {
+        currentTimeProvider.set(STARTUP_TIME)
+        speedometer = new Speedometer(WINDOW_SIZE_IN_SECONDS, currentTimeProvider)
+    })
+
+    it('empty', () => {
+        currentTimeProvider.set('2000-01-02T03:04:05.678Z')
+        expect(speedometer.getRate()).toBe(0)
+    })
+
+    it('get immedatelly', () => {
+        currentTimeProvider.set('2000-01-02T03:04:05.678Z')
+        speedometer.record(100)
+        expect(speedometer.getRate()).toBe(20)
```

Comment:
Would this also work?
```suggestion
        expect(speedometer.getRate()).toBe(100 / WINDOW_SIZE_IN_SECONDS)
```